src/leptonica.rs

```diff
@@ -57,144 +51,68 @@ pub enum PixError {
 pub fn pix_read(path: &Path) -> Result<Pix, PixError> {
     let s = path.to_str().ok_or(PixError::InvalidUtf8Path)?;
     let cs = CString::new(s)?;
-    let pix = unsafe { capi::pixRead(cs.as_ptr()) };
-    // on read errors, leptonica sets pointer to null and prints error message to stderr, so there
-    // is no easy to capture and return the actual message programatically
-    if pix.is_null() {
-        Err(PixError::ReadFrom("file"))
-    } else {
-        Ok(Pix { raw: pix })
+    match leptonica_plumbing::Pix::read(&cs) {
+        Err(leptonica_plumbing::PixReadError()) => Err(PixError::ReadFrom("file")),
+        Ok(raw) => Ok(Pix { raw }),
     }
 }
 
-/// Like pix_read, but redas the image from memory instead of disk
+/// Like pix_read, but reads the image from memory instead of disk
 pub fn pix_read_mem(img: &[u8]) -> Result<Pix, PixError> {
-    let pix = unsafe { capi::pixReadMem(img.as_ptr(), img.len().try_into()?) };
-    if pix.is_null() {
-        Err(PixError::ReadFrom("memory"))
-    } else {
-        Ok(Pix { raw: pix })
+    match leptonica_plumbing::Pix::read_mem(img) {
+        Err(leptonica_plumbing::PixReadMemError::ImageSizeConversion(source)) => {
+            Err(PixError::MemoryTooLarge { source })
+        }
+        Err(leptonica_plumbing::PixReadMemError::NullPtr) => Err(PixError::ReadFrom("memory")),
+        Ok(raw) => Ok(Pix { raw }),
     }
 }
 
-#[derive(Debug, PartialEq)]
-pub struct BoxVal {
-    pub x: i32,
-    pub y: i32,
-    pub w: i32,
-    pub h: i32,
+pub struct Box<R: AsRef<leptonica_plumbing::leptonica_sys::Box>> {
```

Comment:
instead of wrapping the box type here, how about we just reexport the box type from the plumbing crate?

Reply:
Good plan

37957202fe9a0ae68d1c431665bfd8a1d9078f6c

Done here- it cut down on a couple dozen lines of code.